plugin/trino-hive/src/main/java/io/trino/plugin/hive/MergeFileWriter.java

```diff
@@ -141,10 +141,21 @@ public long getMemoryUsage()
     }
 
     @Override
-    public void commit()
+    public RollbackAction commit()
     {
-        deleteFileWriter.ifPresent(FileWriter::commit);
-        insertFileWriter.ifPresent(FileWriter::commit);
+        Optional<RollbackAction> deleteRollbackAction = deleteFileWriter.map(FileWriter::commit);
+        Optional<RollbackAction> insertRollbackAction = insertFileWriter.map(FileWriter::commit);
+        return new RollbackAction(
+                () -> {
+                    // Make sure both writers get rolled back
+                    try {
+                        deleteRollbackAction.ifPresent(RollbackAction::rollback);
+                    }
+                    finally {
+                        insertRollbackAction.ifPresent(RollbackAction::rollback);
```

Comment:
This loses the exception from the delete action. This should really use try-with-resources so that we get both exceptions.